### PR TITLE
DROOLS-7040 Migrate DMN adoc from kie-docs 

### PR DESCRIPTION
**Pull request by tarilabs**
*PLEASE DO NOT MERGE WITH SQUASH* kindly use merge commit in order to preserve migrated doc history.

@mariofusco @tkobayas @evacchi @kaldesai Please review, thanks!
Please notice this is mainly a migration work of the DMN main docs as requested, with minimal adjustments: I wouldn't focus on a content-rewriting at this time, and complete the migration of the Adocs to Antora as requested. I've naturally adjusted parts which are not in scope of 8.x such as kie-server.

**JIRA**: https://issues.redhat.com/browse/DROOLS-7040

**referenced Pull Requests**: none

<details>
<summary>
How to replicate CI configuration locally?
</summary>

Build Chain tool does "simple" maven build(s), the builds are just Maven commands, but because the repositories relates and depends on each other and any change in API or class method could affect several of those repositories there is a need to use [build-chain tool](https://github.com/kiegroup/github-action-build-chain) to handle cross repository builds and be sure that we always use latest version of the code for each repository.
 
[build-chain tool](https://github.com/kiegroup/github-action-build-chain) is a build tool which can be used on command line locally or in Github Actions workflow(s), in case you need to change multiple repositories and send multiple dependent pull requests related with a change you can easily reproduce the same build by executing it on Github hosted environment or locally in your development environment. See [local execution](https://github.com/kiegroup/github-action-build-chain#local-execution) details to get more information about it.
</details>

<details>
<summary>
How to retest this PR or trigger a specific build:
</summary>

- for <b>pull request checks</b>  
  Please add comment: <b>Jenkins retest this</b>

- for a <b>specific pull request check</b>  
  Please add comment: <b>Jenkins (re)run [drools|kogito-runtimes|kogito-apps|kogito-examples] tests</b>

- for a <b>full downstream build</b> 
  - for <b>jenkins</b> job: please add comment: <b>Jenkins run fdb</b>
  - for <b>github actions</b> job: add the label `run_fdb`

- <b>a compile downstream build</b> please  add comment: <b>Jenkins run cdb</b>

- <b>a full production downstream build</b> please add comment: <b>Jenkins execute product fdb</b>

- <b>an upstream build</b> please add comment: <b>Jenkins run upstream</b>

- for <b>quarkus branch checks</b>  
  Run checks against Quarkus current used branch  
  Please add comment: <b>Jenkins run quarkus-branch</b>

- for a <b>quarkus branch specific check</b>  
  Run checks against Quarkus current used branch  
  Please add comment: <b>Jenkins (re)run [drools|kogito-runtimes|kogito-apps|kogito-examples] quarkus-branch</b>

- for <b>quarkus main checks</b>  
  Run checks against Quarkus main branch  
  Please add comment: <b>Jenkins run quarkus-main</b>

- for a <b>specific quarkus main check</b>  
  Run checks against Quarkus main branch  
  Please add comment: <b>Jenkins (re)run [drools|kogito-runtimes|kogito-apps|kogito-examples] quarkus-branch</b>

- for <b>native checks</b>  
  Run native checks  
  Please add comment: <b>Jenkins run native</b>

- for a <b>specific native check</b>  
  Run native checks 
  Please add comment: <b>Jenkins (re)run [drools|kogito-runtimes|kogito-apps|kogito-examples] native</b>

- for <b>mandrel checks</b>  
  Run native checks against Mandrel image
  Please add comment: <b>Jenkins run mandrel</b>

- for a <b>specific mandrel check</b>  
  Run native checks against Mandrel image  
  Please add comment: <b>Jenkins (re)run [drools|kogito-runtimes|kogito-apps|kogito-examples] mandrel</b>
</details>

### CODE REVIEW

**Comment by tarilabs** [comment]
*PLEASE DO NOT MERGE WITH SQUASH* kindly use merge commit in order to preserve migrated doc history.

**Review by tkobayas — APPROVED** [review]
Thanks!

File: drools-docs/src/modules/ROOT/pages/DMN/dmn-decision-tables-con.adoc [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+Decision tables are a popular way of modeling rules and decision logic, and are used in many methodologies (such as DMN) and implementation frameworks (such as Drools).
+
+IMPORTANT: {PRODUCT} supports both DMN decision tables and Drools-native decision tables, but they are different types of assets with different syntax requirements and are not interchangeable. For more information about Drools-native decision tables in {PRODUCT}, see
+ifdef::DM,PAM[]
+{URL_DEVELOPING_DECISION_SERVICES}#assembly-decision-tables[_{SPREADSHEET_DECISION_TABLES}_].
+endif::[]
+ifdef::DROOLS,JBPM,OP[]
+xref:decision-tables-con_decision-tables[].
+endif::[]
```
Comment (tkobayas) [inline drools-docs/src/modules/ROOT/pages/DMN/dmn-decision-tables-con.adoc]: I see you get rid of `ifdef`. Is this a leftover?

Comment (tarilabs) [reply drools-docs/src/modules/ROOT/pages/DMN/dmn-decision-tables-con.adoc]: nice catch, thanks!

**Comment by tarilabs** [comment]
reminder note
*PLEASE DO NOT MERGE WITH SQUASH* kindly use merge commit in order to preserve migrated doc history.

**Comment by evacchi** [comment]
So, do I understand correctly this is "just" moving the original docs to the new docs directory of the drools repo and little else? in that case it should be fine :)

**Comment by evacchi** [comment]
(WTH has just happened sorry)

**Comment by tarilabs** [comment]
@evacchi yes as listed in JIRA, internal memo, and internal gdoc, mainly a migration of the content with some adaptations as described in the internal memo.

To eventually merge, I have already given an headsup to Daniele because I need the unlock/relock of the merge commit.

Hence, the 
reminder note
PLEASE DO NOT MERGE WITH SQUASH kindly use merge commit in order to preserve migrated doc history.

**Comment by tarilabs** [comment]
failure in `Kogito Apps / ubuntu-latest - Java 11 - Maven (pull_request) Failing after 71m` clearly unrelated ([link](https://github.com/kiegroup/drools/runs/7115369382?check_suite_focus=true))
```
kiegroup_kogito_apps/apps-integration-tests/integration-tests-jobs-service/integration-tests-jobs-service-quarkus/integration-tests-jobs-service-quarkus-management/src/test/java/org/kie/kogito/it/jobs/ProcessAsyncIT.java:1 | org.awaitility.core.ConditionTimeoutException:  Assertion condition defined as a lambda expression in org.kie.kogito.it.jobs.BaseProcessAsyncIT 1 expectation failed.
```

**Review by kaldesai — APPROVED** [review]
@tarilabs Looks good to me! Thank you!

**Pull request merged**